kong/api/routes/certificates.lua

```diff
@@ -98,13 +101,129 @@ return {
       })
     end,
 
-
     PUT = function(self, dao_factory, helpers)
-      return crud.put(self.params, dao_factory.ssl_certificates)
+      -- no id present, behaviour should be same as POST
+      if not self.params.id then
+        return create_certificate(self, dao_factory, helpers)
+      end
+
+      --id present in body
+
+      -- check if exists
+      local ssl_cert, err = dao_factory.ssl_certificates:find {
+        id = self.params.id
+      }
+
```

Comment:
No need for line jumps between a function call and its error checking - better keeping them close together to easily browse the code.